### PR TITLE
Replace 'out of capacity' by 'node allocation failure'

### PR DESCRIPTION
**Pull request by bratseth**
This might lead to some temporary glitches during infrastructure upgrade, but I don't think it's worth doing something more complicated.

### CODE REVIEW

File: config-model/src/test/java/com/yahoo/vespa/model/test/VespaModelTester.java [hunk 1]
```diff
@@ -109,7 +109,7 @@ public void useDedicatedNodeForLogserver(boolean useDedicatedNodeForLogserver) {
         this.useDedicatedNodeForLogserver = useDedicatedNodeForLogserver;
     }
 
-    /** Creates a model which uses 0 as start index and fails on out of capacity */
+    /** Creates a model which uses 0 as start index and fails on node allocation */
```
Comment (jonmv) [inline config-model/src/test/java/com/yahoo/vespa/model/test/VespaModelTester.java:112]: I hope it doesn't _always_ fail on node allocation? 😜

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/RunSerializer.java [hunk 2]
```diff
@@ -377,7 +377,8 @@ static String valueOf(RunStatus status) {
     static RunStatus runStatusOf(String status) {
         switch (status) {
             case "running"                    : return running;
-            case "outOfCapacity"              : return outOfCapacity;
+            case "outOfCapacity"              : return nodeAllocationFailure;  // TODO: Remove after March 2022
```
Comment (jonmv) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/RunSerializer.java:380]: We don't re-write historic runs, so this needs to stay, unless we start re-writing all of those regularly as well.

Comment (bratseth) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/RunSerializer.java:380]: When do they expire?

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/InternalStepRunner.java [hunk 3]
```diff
@@ -812,7 +812,7 @@ private void updateConsoleNotification(Run run) {
             case success:
                 controller.notificationsDb().removeNotification(source, Notification.Type.deployment);
                 return;
-            case outOfCapacity:
+            case nodeAllocationFailure:
                 if ( ! run.id().type().environment().isTest()) updater.accept("lack of capacity. Please contact the Vespa team to request more!");
```
Comment (jonmv) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/InternalStepRunner.java:816]: This should dalso be updated, like the mail notification.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/RunSerializer.java (hunk 2 above)
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/RunSerializer.java:380]: When they're more than 65 runs old, or somehting, and they're also no longer the "first failing" or "lasts successful" run. Potentially never. I guess we should re-write these runs ...

Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/RunSerializer.java:380]: I can do that in a differnet PR.

Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/RunSerializer.java:380]: Never mind I'm just bad at reading code. We update these as well.

Comment (bratseth) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/RunSerializer.java:380]: Thanks. I'll remove the date for now.

**Review by jonmv — APPROVED** [review]

**Comment by hmusum** [comment]
Users of `OutOfCapacityException` in internal repo also need to be updated (build is broken now)

**Pull request merged**